app/grandchallenge/components/serializers.py

```diff
@@ -26,32 +30,130 @@ class SimpleImageSerializer(serializers.ModelSerializer):
     # name is needed
     class Meta:
         model = Image
-        fields = (
-            "pk",
-            "name",
-        )
+        fields = ("pk", "name")
 
 
-class ComponentInterfaceValueSerializer(serializers.ModelSerializer):
-    # Serializes images in place rather than with hyperlinks for internal usage
-    image = SimpleImageSerializer()
-    interface = ComponentInterfaceSerializer()
+class ComponentInterfaceValuePostSerializer(serializers.ModelSerializer):
+    """
+    Serializes images with hyperlinks for external usage
+    Expects interface_slug, to allow creating a ComponentInterfaceValue with an
+    existing ComponentInterface
+    """
+
+    image = serializers.HyperlinkedRelatedField(
+        queryset=Image.objects.all(),
+        view_name="api:image-detail",
+        required=False,
+    )
+
+    interface_slug = serializers.CharField(write_only=True)
+    interface = ComponentInterfaceSerializer(read_only=True)
+    upload_pk = serializers.CharField(write_only=True, required=False)
 
     class Meta:
         model = ComponentInterfaceValue
         fields = [
+            "interface_slug",
             "interface",
             "value",
             "file",
             "image",
             "pk",
+            "upload_pk",
         ]
 
+    def validate(self, attrs):  # noqa: C901
+        interface = ComponentInterface.objects.get(
+            slug=attrs["interface_slug"]
+        )
```

Comment:
I think this will throw ModelDoesNotExist if no ComponentInterface exists with this `interface_slug`. Probably better to raise a ValidationError if that is the case.